planet/clients/orders.py

```diff
@@ -313,6 +338,26 @@ async def download_order(self,
                                       progress_bar=progress_bar)
             for location in locations
         ]
+        if checksum:
```

Comment:
This only works if you're expecting `checksum` to not be `None`, but your code looks for `checksum` to either be `MD5` or `SHA256`, so you can replace this with `checksum == 'md5' or 'sha256'` for it to work.